### PR TITLE
Console log message to prevent self xss

### LINKED ISSUE #15: Security: Add Self XSS Warning to Dev Console
Add a warning in the Javascript console similar to this example from www.facebook.com
image

### PR DESCRIPTION
**Pull request by kloon**
This adds a developer console message informing the user about the possibility of a self XSS attack if they were asked to enter a value in the console.

Image of message

This closes #15, it could do with additions like translating the string based on locale and checking if the console support colors and styles, the latter only possible via user agent checking, unfortunately.

Could also be useful to setup a page on wordpress.com explaining self XSS and then linking to that page.

### CODE REVIEW

**Comment by gziolo** [comment]
There is also #11625 opened with a similar change. We should land one of the solutions. @dmsnell you reviewed the other PR, so it would be nice to finally one of the solutions.

**Comment by spen** [comment]
What do you think about a "We're hiring" nudge in this?
Something with the sentiment of: "Don't do continue if you don't know what you're doing, but if you do come and join in"

It could just confuse people since it gives mixed messages, so it might not be a great idea 🤷‍♂️

**Comment by kloon** [comment]
@spen I thought about adding the hiring message but figured it would not go well with the message about self XSS. There's also a x-hacker header with details about hiring on all wp.com pages `x-hacker:If you're reading this, you should visit automattic.com/jobs and apply to join the fun, mention this header.`

**Comment by gziolo** [comment]
Agree, it would be nice to get thumbs up from @nb or @ehg before merging this change.

**Comment by spen** [comment]
Thanks @kloon, I think that make sense so as not to confuse folks :)

Another consideration might be to suppress the message in non-production environments - I'm not sure the best path for doing so in our jade templates, and not sure if this would cause more problems than it solves.

**Comment by gziolo** [comment]
@hoverduck, does it solve #15 as you expected?

**Comment by hoverduck** [comment]
@gziolo - Yes, this generally solves #15, although @allendav was the original author...I only ported the issue over from the pre-oss repo :smile: 

A couple issues, though -
It works fine in Chrome, but I checked IE11 and it doesn't show up.  Those were the only browsers I looked at.

And just a nit-pick, but I'd update it to say "This browser **feature** is intended for developers" (singular) instead of "This browser **features** is intended for developers" (plural)

File: server/pages/index.jade [hunk 1]
```diff
@@ -166,5 +166,17 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
 				script(src=urls[ chunk + '-min' ])
 		script(type='text/javascript')!='window.AppBoot();'
 
+		script.
+			(function() {
+				if ( window.console ) {
+					console.log( "%cSTOP!", "color:#f00;font-size:xx-large" );
+					console.log(
+						"%cThis browser features is intended for developers. " +
```
Comment (allendav) [inline server/pages/index.jade]: Grammar nit: `features` should probably be `feature`

**Comment by dmsnell** [comment]
@folletto or @melchoyce or @adambbecker or someone could we get some design review of the text here as this will be public-facing communication from Automattic?

a part of me feels like it's too dread-serious for most of our other communication.

**Comment by folletto** [comment]
Let's just tag Editorial ;)

**Comment by michelleweber** [comment]
It's okay to be serious when there are potentially bad consequences! That said, I would tweak a bit to give people a little more context about *why* this is potentially bad:

Wait! This browser feature runs code that can alter your website or its security, and is intended for developers. If you've been told to copy and paste something here to enable a feature, someone may be trying to compromise your account. Please make sure you understand the code and trust the source before adding anything here.

**Comment by dmsnell** [comment]
> Please make sure you understand the code and trust the source before adding anything here.

Thanks @michelleweber! I like how your wording acknowledges that the console can still be useful (we're not trying to forbid people from using it) and that some people will have the right/skill/know-how to get in there and dig around (while still warning off people who may be mislead by something they read online).

**Comment by gziolo** [comment]
Yes, feel free to merge 👍

**Comment by kwight** [comment]
This does feel very distracting and unnecessary in the `development` environment; #19355 to remove.

**Pull request merged**